service/middleware.go

```diff
@@ -228,8 +229,35 @@ func createProxyRequestMiddleware(next http.Handler, config config.Config, servi
 				serviceLogger.Trace().Msg("request body is empty, skipping before request interceptors")
 			}
 
-			// proxy request to backend origin servers
-			proxy.ServeHTTP(lrw, r)
+			isCached := cachemdw.IsRequestCached(r.Context())
+			response := r.Context().Value(cachemdw.ResponseContextKey)
+			typedResponse, ok := response.([]byte)
+
+			// if request is cached and response is present in context - serve the request from the cache
+			// otherwise proxy to the actual backend
+			if isCached && ok {
+				serviceLogger.Logger.Trace().
+					Str("method", r.Method).
+					Str("url", r.URL.String()).
+					Str("host", r.Host).
```

Comment:
these top 3 will be the same for the evm json rpc api (all http POST requests to the same root url, can we log the evm method here instead please

Reply:
@galxy25 
fixed here: https://github.com/Kava-Labs/kava-proxy-service/pull/39/commits/853b7266998ac1395a020030550655498b73b45e

lmk if implementation is reasonable